src/parser/parse.ts

```diff
@@ -191,14 +192,24 @@ export class Parser<T extends ParserInput, TFlags extends OutputFlags<T['flags']
           flags[token.flag] = true
         }
 
-        // eslint-disable-next-line no-await-in-loop
-        flags[token.flag] = await flag.parse(flags[token.flag], this.context, flag)
+        try {
+          // eslint-disable-next-line no-await-in-loop
+          flags[token.flag] = await flag.parse(flags[token.flag], this.context, flag)
+        } catch (error: any) {
+          throw new Error(`Parsing ${token.input} \n\t${error.message}`)
```

Comment:
This should be a `CLIParseError`: https://github.com/oclif/core/blob/main/src/parser/errors.ts#L27

Reply:
Just pushed a change that abstracted this out into a new error type called `FailedFlagParseError`